parla/ldevice.py

```diff
@@ -341,3 +342,58 @@ def wrapper(*args, memory, device):
         def wrapper(*args, memory, device):
             return memory(data(*args))
     return wrapper
+
+
+class Coordinator():
```

Comment:
This is a bad name. I'm not at all sure what's going on here. I think `Coordinator` is actually a partitioned array abstraction for arrays that are split across devices. And if so, let's call it `PartitionedNDArray`.

Also please document the semantics of the class and it's methods in doc strings. It's not clear exactly what each function is doing.

* What is the location of the the values returned by `__getitem__` and `__setitem__`?
* What happens if I assign into the result of `__getitem__` (e.g., `local_array = partitioned_array[x]; local_array[:] = ...;`)? Is this write local only? (it seems like it is.)

I don't care at the moment what the semantics are. Just please write down what you implemented.

Reply:
Totally agreed this is a bad name 😛. I was actually trying to be forward-looking - this wrapper should also do some bookkeeping of where the parts have been and thus "coordinate"s the movement.

Documentation is important since I have assumptions on the programming paradigm. Will focus on clarifying what I'm/'ll be doing.

Reply:
As a general rule classes should be named based on what abstraction they provide, not how they implement that abstraction. So the class is a partitioned nd array, the underlying coordination doesn't matter to a user of the class.